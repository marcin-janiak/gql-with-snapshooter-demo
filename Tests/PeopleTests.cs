using System;
using System.Threading.Tasks;
using API;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;
using Xunit;

namespace Tests
{
    public class PeopleTests
    {
        private readonly IRequestExecutor _executor;
        private readonly ServiceProvider _serviceProvider;

        public PeopleTests()
        {
            var services = new ServiceCollection();
            services.AddDbContext<PeopleContext>(options => { options.UseInMemoryDatabase("PeopleContext"); });

            services.AddGraphQLCore();
            
            _serviceProvider = services.BuildServiceProvider();

            _executor = new SchemaBuilder().AddQueryType<Query>()
                .AddMutationType<Mutation>()
                .AddServices(_serviceProvider)
                .Create()
                .MakeExecutable();
        }

        [Fact]
        public async Task Test()
        {
            var context = _serviceProvider.GetService<PeopleContext>()!;
            context.Add(new Person
                {
                    Id = new Guid("553A403C-CD68-4C7D-A2EC-FEA6428CB7C9"),
                    Name = "Test name"
                });
            
            await context.SaveChangesAsync();

            IReadOnlyQueryRequest request =
                QueryRequestBuilder.New()
                    .SetQuery(
                        @"
query getPeople {
  allPeople {
    id
    name
    supervisorId
  	supervisor {
      id
      name
      supervisorId
    }
  }
}")
                    .SetServices(_serviceProvider)
                    .Create();


            var result = await _executor.ExecuteAsync(request);
            
            result.MatchSnapshot();
        }
    }
}