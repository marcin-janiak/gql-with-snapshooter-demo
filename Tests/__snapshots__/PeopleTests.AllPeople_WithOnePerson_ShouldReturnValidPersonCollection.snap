{
  "Label": null,
  "Path": null,
  "Data": {
    "allPeople": [
      {
        "id": "553a403ccd684c7da2ecfea6428cb7c9",
        "name": "Test name",
        "supervisorId": null,
        "supervisor": null
      }
    ]
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
